.github/workflows/ci.yaml

```
@@ -71,6 +71,22 @@ jobs:
               - 'images/nginx-1.25/**'
             docs:
               - '**/*.md'
+            lua:
+              - '**/*.lua'
+
+  lua-lint:
+    runs-on: ubuntu-latest
+    needs: changes
+    if: |
+      (needs.changes.outputs.lua == 'true') || ${{ github.event.workflow_dispatch.run_e2e == 'true' }}
```

```suggestion
    if: fromJSON(needs.changes.outputs.lua) || fromJSON(github.event.workflow_dispatch.run_e2e)
```

Have you tried this?

---

never tried. but then wouldn't it be better to change all of the ifs for this approach on a follow up PR?